### PR TITLE
added code to provide output file basename

### PR DESCRIPTION
**Pull request by tboser**
added code to provide output file basename$
fixed command to create docker tags

worth noting 
-in input json file the basename given needs to match basename on output files or CWL will throw an exception.
-removed "docker rmi ${name}:${short_tag}" which may cause issues if you want to replace a docker image?

### CODE REVIEW

**Comment by tboser** [comment]
@jvivian

File: docker/wrapper.py [hunk 1]
```diff
@@ -56,8 +56,25 @@ def call_pipeline(mount, args):
         else:
             log.info('Flag "--no-clean" was used, therefore {} was not deleted.'.format(work_dir))
 
+    #DEBUG !!!
+    print("DEBUG!!!!! output dir is "+mount+" and files are:")
+    print(os.listdir(mount))
+    fail_files = [fail_file for fail_file in os.listdir(mount) if fail_file.startswith("FAIL.") and fail_file.endswith(".tar.gz")]
```
Comment (jvivian) [inline docker/wrapper.py]: `fail_files = glob.glob(os.path.join(mount, 'FAIL*'))`
or
`fail_files = [os.path.join(mount, x) for x in os.listdir(mount) if x.startswith('FAIL')]`

PyCharm's PEP is lines < 120 characters.

File: docker/wrapper.py [hunk 2]
```diff
@@ -56,8 +56,25 @@ def call_pipeline(mount, args):
...
+    print("fail files:", ' '.join(fail_files))
+    cmd = ["mv", mount+"/"+fail_file, mount+"/"+fail_file[len("FAIL."):]]
```
Comment (jvivian) [inline docker/wrapper.py]: - This line is creating `cmd` based off a temporary variable, this is broken and will not do what you want. 
- Missing PEP convention for spaces around operators.
- Don't combine file paths with strings and slashes, it's not OS agnostic. Use `os.path.join`.
- Replace the string slice with `fail_file.lstrip('FAIL.')`
- If you make the `glob` change as suggested above, `mount` will already be part of the path for the elements in `fail_files`.

File: docker/wrapper.py [hunk 3]
```diff
@@ -56,8 +56,25 @@ def call_pipeline(mount, args):
...
+    for fail_file in fail_files:
+        print("moving "+mount+"/"+fail_file+" to "+mount+"/"+fail_file[len("FAIL."):])
```
Comment (jvivian) [inline docker/wrapper.py]: Use logger. Fix for formatting / PEP.

File: docker/wrapper.py [hunk 4]
```diff
@@ -56,8 +56,25 @@ def call_pipeline(mount, args):
...
+        try:
+            subprocess.check_call(cmd) 
+        except subprocess.CalledProcessError as e:
+            print(e.message, file=sys.stderr)
+        except Exception as e:
+            print("\nERROR: FAIL file mv exception information:" + str(e), file=sys.stderr)
+    #END DEBUG !!!!
```
Comment (jvivian) [inline docker/wrapper.py]: Remove.

File: docker/wrapper.py [hunk 5]
```diff
@@ -98,10 +115,13 @@ def fileURL(sample):
...
 
-def getSampleName(sample):
-    name = os.path.basename(sample).split('.')[0]
-    if name.endswith('R1') or name.endswith('R2'):
-        return name[:-2]
+def getSampleName(sample, output_basename):
+    if output_basename:
+        name = output_basename
+    else:
+        name = os.path.basename(sample).split('.')[0]
```
Comment (jvivian) [inline docker/wrapper.py]: Either this function needs to go and `output_basename` becomes a required variable (recommended), or you improve the heuristic for deriving a name form the sample string.  Filenames for samples can have dots in them that are not just for the extension.

File: docker/wrapper.py (hunk 1 above)
Comment (jvivian) [inline docker/wrapper.py]: `fail_files = glob.glob(os.path.join(mount, '*FAIL'))`

Pycharm PEP < 120 characters per line

**Comment by jvivian** [comment]
Will close pending #58.

**Comment by jvivian** [comment]
Closing in lieu of #58

**Pull request closed without merging**